data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/datatype/impl/TemporalTypeHandler.java

```diff
@@ -4,22 +4,133 @@
+    private static final String MYSQL_DATETIME_FORMAT = "yyyy-MM-dd HH:mm:ss[.SSSSSS]";
+
+    // Thread-safe formatters
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern(MYSQL_DATE_FORMAT);
+    private static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern(MYSQL_TIME_FORMAT);
+    private static final DateTimeFormatter DATETIME_FORMATTER = DateTimeFormatter.ofPattern(MYSQL_DATETIME_FORMAT);
 
     @Override
-    public String handle(final MySQLDataType columnType, final String columnName, final Object value,
-                         final TableMetadata metadata) {
-        // Date and Time types
-        switch (columnType) {
-            // TODO: Implement the transformation
-            case DATE:
-            case TIME:
-            case TIMESTAMP:
-            case DATETIME:
-            case YEAR:
-                return value.toString();
-            default:
-                throw new IllegalArgumentException("Unsupported temporal data type: " + columnType);
+    public Long handle(final MySQLDataType columnType, final String columnName, final Object value,
+                       final TableMetadata metadata) {
+        if (value == null) {
+            return null;
+        }
+
+        final String strValue = value.toString().trim();
+
+        try {
+            switch (columnType) {
+                case DATE:
+                    return handleDate(strValue);
+                case TIME:
+                    return handleTime(strValue);
+                case DATETIME:
+                case TIMESTAMP:
+                    return handleDateTime(strValue);
+                case YEAR:
+                    return handleYear(strValue);
+                default:
+                    throw new IllegalArgumentException("Unsupported temporal data type: " + columnType);
+            }
+        } catch (Exception e) {
+            throw new IllegalArgumentException(
+                    String.format("Failed to parse %s value: %s", columnType, strValue), e);
+        }
+    }
+
+    private Long handleTime(final String timeStr) {
+        try {
+            // Try parsing as Unix timestamp first
+            final Long timeEpoch = parseDateTimeStrAsEpoch(timeStr);
+            if (timeEpoch != null) return timeEpoch;
+
+            final LocalTime time = LocalTime.parse(timeStr, TIME_FORMATTER);
+            // Combine with date from EPOCH
+            return time.atDate(LocalDate.EPOCH)
+                    .toInstant(ZoneOffset.UTC)
+                    .toEpochMilli();
+        } catch (DateTimeParseException e) {
+            throw new IllegalArgumentException("Invalid time format: " + timeStr, e);
+        }
+    }
+
+    private Long handleDate(final String dateStr) {
+        try {
+            // Try parsing as Unix timestamp first
+            final Long dateEpoch = parseDateTimeStrAsEpoch(dateStr);
+            if (dateEpoch != null) return dateEpoch;
```

Comment:
Is the return value here epoch in millis or in days, since "Date from binlog is long value representing days since epoch"?

Reply:
It's Unix time (number of seconds that have elapsed since 00:00:00 (UTC), Thursday, 1 January 1970, not counting leap seconds)

Reference:
https://github.com/shyiko/mysql-binlog-connector-java/blob/dd710a5466381faa57442977b24fceff56a0820e/src/main/java/com/github/shyiko/mysql/binlog/event/deserialization/AbstractRowsEventDataDeserializer.java#L511-L533

https://github.com/shyiko/mysql-binlog-connector-java/blob/dd710a5466381faa57442977b24fceff56a0820e/src/main/java/com/github/shyiko/mysql/binlog/event/deserialization/AbstractRowsEventDataDeserializer.java#L248-L270

Reply:
In that case, we need to convert it to epoch millis to be consistent with the below conversion for export data?

Reply:
yeah, will make change to configure BinlogClient with `EventDeserializer.CompatibilityMode.DATE_AND_TIME_AS_LONG_MICRO`

Reply:
Updated documentation